pkg/apis/certmanager/v1alpha1/types_certificate.go

```diff
@@ -57,6 +57,12 @@ type CertificateSpec struct {
 	// Organization is the organization to be used on the Certificate
 	Organization []string `json:"organization,omitempty"`
 
+	// Certificate default Duration
+	Duration metav1.Duration `json:"duration,omitempty"`
+
+	// Certificate renew before expiration duration
+	RenewBefore metav1.Duration `json:"renewBefore,omitempty"`
```

Comment:
The `omitempty` struct tag does not do anything for non-pointer structs. I think we may need to consider making both of these fields pointers, so that they are excluded from output when not set, and also to make it easier to compare to the zero value.

That said, I'm happy to merge this now and open an issue to verify the behaviour is as we want ahead of cutting v0.6.0.

Reply:
It does work, if I remember correctly, if I just leave off the Duration fields, or is that not what we're talking about? I have no problem changing it to a pointer though and checking for nil makes more sense logically than checking for 0.

Reply:
Chatting to people in sig-api-machinery, it sounds like this field should be a pointer as it is optional. Once we have proper defaulting with a mutating webhook, we can consider changing this back (and defaulting it to the real default value).

If it isn't too much work to switch now, that'd be great!